core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1281,6 +1280,45 @@ func (o *evmTxStore) UpdateEthTxAttemptInProgressToBroadcast(etx *EvmTx, attempt
 	})
 }
 
+// recoverFromTxAttemptsHashConflict recovers from an idx_eth_tx_attempts_hash constraint violation.
+// If a replay was triggered while unconfirmed transactions were pending, they will be marked as fatal_error => abandoned.
+// In this case, we must remove the abandoned attempt from eth_tx_attempts before replacing it with a new one.  In any other
+// case, we uphold the constraint, leaving the original tx attempt as-is and returning the constraint violation error.
+//
+// Note:  the record of the original abandoned transaction will remain in eth_txes, only the attempt is replaced.  (Any receipt
+// associated with the abandoned attempt would also be lost, although this shouldn't happen since only unconfirmed transactions
+// can be abandoned.)
+func recoverFromTxAttemptsHashConflict(attempt *EvmTxAttempt, tx pg.Queryer, dupErr error, query string, args ...interface{}) (err error) {
+	var duplicateTx DbEthTx
+	defer func() {
+		if err != nil {
+			err = multierr.Append(dupErr, err)
```

Comment:
nit/ std lib is sufficient now
```suggestion
			err = errors.Join(dupErr, err)
```